uvicorn/loops/asyncio.py

```diff
@@ -5,7 +5,7 @@
 logger = logging.getLogger("uvicorn.error")
 
 
-def asyncio_setup(reload: bool = False) -> None:  # pragma: no cover
-    if sys.version_info >= (3, 8) and sys.platform == "win32" and reload:
+def asyncio_setup(use_subprocess: bool = False) -> None:  # pragma: no cover
+    if sys.version_info >= (3, 8) and sys.platform == "win32" and use_subprocess:
         logger.warning("The --reload flag should not be used in production on Windows.")
```

Comment:
This warning is wrong now, because we can reach this line with the `--workers`.

Reply:
yep indeed, so the use_subprocess flag is cool on the surface but does not allow to discriminate between workers usage and reload usage so I think I'm gonna revert to having both in the signature despite your comment here (https://github.com/encode/uvicorn/pull/1454#issuecomment-1118221327), what you think ? or you get an idea as to detect at running time the fact we're in a subprocess ?

Reply:
The thing is that the warning itself doesn't make sense to me. If uvicorn is running under `--reload` and we show the warning, it's not needed, because it's not in production anyway... How do we even know that is being used in production?

Reply:
ok let's dig why we put this in the first place, cant remember

Reply:
it was introduced in #1257

Reply:
If we just want to add a warning message because we want to be sure users don't use reload in production, that should be in the main.py, not on the loop setup.

Reply:
ok so let's keep use_subprocess and remove the warning ?

Reply:
oh and f*** windows

Reply:
> ok so let's keep use_subprocess and remove the warning ?

Yes 👍 

If we still want the warning in the future, we can add it before those lines:
https://github.com/encode/uvicorn/blob/b1a45821a2ab916b4428bdcb9eba4a4a0cc8fb74/uvicorn/main.py#L552-L557

I think the warning is acceptable, as we want to be sure the user doesn't make the mistake of running `--reload` in production. But that's for another PR/discussion.

Reply:
should be good for another round of review @Kludex